pkgs/top-level/config.nix

```diff
@@ -26,7 +26,12 @@ let
     '';
   });
 
-  options = {
+  mkOverrides = args: mkMassRebuild ({
+    type = types.functionTo (types.lazyAttrsOf (types.uniq types.unspecified));
+    default = super: {};
```

Comment:
Actually I don't think we need all these new types by just not supporting setting `packageOverrides` multiple times, which is totally okay because this wasn't supported previously either. `packageOverrides`-like ways are kind of deprecated anyways too, we don't need to extend their support by giving new features. If people want to have overlay-like behavior they should use overlays, which package sets can add support for too (e.g. I'm trying to get python to support overlays right now in https://github.com/NixOS/nixpkgs/pull/67422). In addition, this type introduces yet another slightly different way of overriding packages, different from any other ways. I think we have enough ways of overriding things already, this will only make it more confusing.

So in short, I think this can be `types.uniq types.unspecified` without worry, then the controversial `functionTo` and `lazyAttrsOf` can be dropped from this PR.